GitUI/CommandsDialogs/FormBrowse.cs

```diff
@@ -946,12 +946,17 @@ private void UpdateStashCount()
         {
             if (AppSettings.ShowStashCount)
             {
-                ThreadHelper.JoinableTaskFactory.RunAsync(() =>
+                ThreadHelper.JoinableTaskFactory.RunAsync(async () =>
                 {
-                    return AsyncLoader.DoAsync(() => Module.GetStashes().Count,
-                        (result) => toolStripSplitStash.Text = string.Format(_stashCount.Text, result,
-                            result != 1 ? _stashPlural.Text : _stashSingular.Text));
-                });
+                    await TaskScheduler.Default.SwitchTo(alwaysYield: true);
```

Comment:
We are on UI thread here, can we drop `alwaysYield: true`?

Reply:
➡️ Yes, `alwaysYield` only has an effect if you are already on the target thread. I'll update both cases for this PR and add assertions to the calling method so the current thread is clear.